src/Controller/MediaController.php

```diff
@@ -61,14 +61,16 @@ public function downloadAction($id, $format = MediaProviderInterface::FORMAT_REF
             throw new NotFoundHttpException(sprintf('unable to find the media with the id : %s', $id));
         }
 
-        if (!$this->get('sonata.media.pool')->getDownloadSecurity($media)->isGranted($media, $this->getCurrentRequest())) {
+        $request = $this->getCurrentRequest();
+
+        if (!$this->get('sonata.media.pool')->getDownloadSecurity($media)->isGranted($media, $request)) {
```

Comment:
Why this change?

Reply:
Not necessary, the thing is that I first changed the signature of actions to add `Request $request`  and removed `getCurrentRequest` and then to respect BC I removed the parameter again and instead of changing back to `$this->getCurrentRequest()` everywhere I reused the variable. I can change it back.

Reply:
Please do :pray: